// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.

syntax = "proto2";

package safari_sdk.protos.logging;

option java_multiple_files = true;
option java_outer_classname = "PolicyTypeProto";

// Proto to respresent the policy type of an episode.
// Next ID: 7
enum PolicyType {
  POLICY_TYPE_UNSPECIFIED = 0;
  // Policy type for teleoperated data collection.
  POLICY_TYPE_ROBOT_TELEOPERATION = 1;
  // Policy type for evaluations.
  POLICY_TYPE_ROBOT_EVALUATION = 2;
  // Policy type for written policies.
  POLICY_TYPE_SCRIPTED = 3;
  // Policy type for human interventions.
  POLICY_TYPE_INTERVENTION = 4;
  // Policy type for orchestrated collection.
  POLICY_TYPE_ORCHESTRATED_COLLECTION = 5;
  // Policy type for orchestrated evaluation.
  POLICY_TYPE_ORCHESTRATED_EVALUATION = 6;
}
